illumos-utils/src/zfs.rs

```diff
@@ -244,3 +244,26 @@ impl Zfs {
         Ok(value.to_string())
     }
 }
+
+/// Returns all datasets managed by Omicron
+pub fn get_all_omicron_datasets() -> anyhow::Result<Vec<String>> {
```

Comment:
This was pulled exactly from `omicron-package` -- but the sled-reset command wants to use it too.